### ISSUE TITLE
[BUG] Concurrent Map Access in TTLSet Causing Fatal Error

### ISSUE THREAD

**Issue by camera-2018** [hdu-dn11/proxyScan#2]
```bash
2024/08/08 01:18:59 [+] xxx.180:20172
2024/08/08 01:18:59 [+] xxx.180:20170
2024/08/08 01:18:59 [+] xxx.180:7890
2024/08/08 01:18:59 wait for tcp scan.
fatal error: concurrent map iteration and map write
2024/08/08 01:18:59 [+] xxx.183:20170
2024/08/08 01:18:59 [+] xxx.184:20171
2024/08/08 01:18:59 [+] xxx.185:10809
2024/08/08 01:18:59 [+] xxx.190:20170
2024/08/08 01:18:59 [+] xxx.191:20172
2024/08/08 01:18:59 [+] xxx.187:7893

goroutine 1 [running]:
github.com/hdu-dn11/proxyScan/utils.(*TTLSet[...]).Wait(0x10c8480)
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/utils/ttlset.go:55 +0xe7
github.com/hdu-dn11/proxyScan/scan/tcpscanner/pcap.(*Scanner).End(0xc000600360)
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/scan/tcpscanner/pcap/pcap.go:47 +0x2a
github.com/hdu-dn11/proxyScan/scan.(*Scanner).ScanSocks5(0xc00091de88, {0xc0006dc9e0, 0x1, 0x1}, {0xc000144180, 0x9, 0x10})
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/scan/scan.go:87 +0x342
github.com/hdu-dn11/proxyScan/cmd/cli.Cli()
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/cmd/cli/cli.go:75 +0x82c
main.main()
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/main.go:12 +0xf

goroutine 18 [select]:
github.com/yaklang/yaklang/common/utils/arptable.AutoRefresh.func1()
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/arptable/arp.go:27 +0x6d
created by github.com/yaklang/yaklang/common/utils/arptable.AutoRefresh in goroutine 1
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/arptable/arp.go:25 +0x37

goroutine 19 [select]:
github.com/yaklang/yaklang/common/utils.(*CacheWithKey[...]).startExpirationProcessing(0x10d99a0)
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:228 +0x278
created by github.com/yaklang/yaklang/common/utils.NewTTLCacheWithKey[...] in goroutine 1
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:419 +0x1a7

goroutine 20 [select]:
github.com/yaklang/yaklang/common/utils.(*CacheWithKey[...]).startExpirationProcessing(0x10d9bc0)
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:228 +0x278
created by github.com/yaklang/yaklang/common/utils.NewTTLCacheWithKey[...] in goroutine 1
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:419 +0x1a7

goroutine 21 [select]:
github.com/yaklang/yaklang/common/utils.(*CacheWithKey[...]).startExpirationProcessing(0x10d9580)
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:228 +0x278
created by github.com/yaklang/yaklang/common/utils.NewTTLCacheWithKey[...] in goroutine 1
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:419 +0x1a7

goroutine 22 [runnable]:
math/big.nat.montgomery({0xc000469e60?, 0x10?, 0x24?}, {0xc0007d2500?, 0x10?, 0x26?}, {0xc000469560?, 0x10?, 0x24?}, {0xc0000d4dc0, ...}, ...)
        /usr/local/go/src/math/big/nat.go:216 +0x558
math/big.nat.expNNMontgomery({0xc00046e0a0, 0x1e4dac0?, 0x14}, {0xc00046e000?, 0x42bb9c?, 0x1e4dac0?}, {0xc000457ea0, 0x10, 0x41615e?}, {0xc0000d4dc0, ...})
        /usr/local/go/src/math/big/nat.go:1269 +0x9f5
math/big.nat.expNN({0x0?, 0x14?, 0x17be2f8?}, {0xc00046e000, 0x10, 0x14}, {0xc000457ea0, 0x10, 0x14?}, {0xc0000d4dc0, ...}, ...)
        /usr/local/go/src/math/big/nat.go:989 +0x37d
math/big.nat.probablyPrimeMillerRabin({0xc0000d4dc0, 0x10, 0x14}, 0x15, 0x1)
        /usr/local/go/src/math/big/prime.go:106 +0x5d5
math/big.(*Int).ProbablyPrime(0xc00030ea60, 0x14)
        /usr/local/go/src/math/big/prime.go:78 +0x213
crypto/rand.Prime({0x10b8f40, 0xc00009ada0}, 0x400)
        /usr/local/go/src/crypto/rand/util.go:55 +0x1e6
crypto/rsa.GenerateMultiPrimeKey({0x10b8f40, 0xc00009ada0}, 0x2, 0x800)
        /usr/local/go/src/crypto/rsa/rsa.go:396 +0x745
crypto/rsa.GenerateKey(...)
        /usr/local/go/src/crypto/rsa/rsa.go:276
github.com/yaklang/yaklang/common/utils/tlsutils.SelfSignCACertificateAndPrivateKey({0xf717cd, 0x4}, {0xc000053d90, 0x4, 0x4?})
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/tlsutils/sign.go:93 +0x4b2
github.com/yaklang/yaklang/common/utils/tlsutils.GenerateSelfSignedCertKeyWithCommonNameEx({0xf717cd, 0x4}, {0xf717cd, 0x4}, {0xf7c345?, 0x9?}, {0x0, 0x0, 0x0?}, {0x0, ...}, ...)
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/tlsutils/tls.go:199 +0x425
github.com/yaklang/yaklang/common/utils/tlsutils.GenerateSelfSignedCertKeyWithCommonNameWithPrivateKeyWithOrg(...)
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/tlsutils/tls.go:181
github.com/yaklang/yaklang/common/utils/tlsutils.GenerateSelfSignedCertKeyWithCommonName({0xf717cd?, 0x0?}, {0xf7c345?, 0x0?}, {0x0?, 0x0?, 0x0?}, {0x0, 0x0, 0x0})
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/tlsutils/tls.go:163 +0x74
github.com/yaklang/yaklang/common/utils/tlsutils.init.0.func1()
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/tlsutils/tls.go:168 +0x4d
github.com/yaklang/yaklang/common/utils.RegisterDefaultTLSConfigGenerator.func1()
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/os_utils.go:362 +0x1b
sync.(*Once).doSlow(0x0?, 0x0?)
        /usr/local/go/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
        /usr/local/go/src/sync/once.go:65
created by github.com/yaklang/yaklang/common/utils.RegisterDefaultTLSConfigGenerator in goroutine 1
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/os_utils.go:361 +0xa5

goroutine 23 [select]:
github.com/yaklang/yaklang/common/utils.(*CacheWithKey[...]).startExpirationProcessing(0x10d8f20)
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:228 +0x278
created by github.com/yaklang/yaklang/common/utils.NewTTLCacheWithKey[...] in goroutine 1
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:419 +0x1a7

goroutine 24 [select]:
github.com/yaklang/yaklang/common/utils.(*CacheWithKey[...]).startExpirationProcessing(0x10d9140)
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:228 +0x278
created by github.com/yaklang/yaklang/common/utils.NewTTLCacheWithKey[...] in goroutine 1
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:419 +0x1a7

goroutine 25 [select]:
github.com/yaklang/yaklang/common/utils.(*CacheWithKey[...]).startExpirationProcessing(0x10d9140)
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:228 +0x278
created by github.com/yaklang/yaklang/common/utils.NewTTLCacheWithKey[...] in goroutine 1
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:419 +0x1a7

goroutine 29 [select]:
github.com/yaklang/yaklang/common/utils.(*CacheWithKey[...]).startExpirationProcessing(0x10d8ae0)
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:228 +0x278
created by github.com/yaklang/yaklang/common/utils.NewTTLCacheWithKey[...] in goroutine 1
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:419 +0x1a7

goroutine 30 [select]:
github.com/yaklang/yaklang/common/utils.(*CacheWithKey[...]).startExpirationProcessing(0x10d9360)
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:228 +0x278
created by github.com/yaklang/yaklang/common/utils.NewTTLCacheWithKey[...] in goroutine 1
        /root/go/pkg/mod/github.com/yaklang/yaklang@v1.3.3/common/utils/cache.go:419 +0x1a7

goroutine 31 [runnable]:
github.com/hdu-dn11/proxyScan/utils.NewCollector[...].func1()
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/utils/collect2slice.go:15 +0x69
created by github.com/hdu-dn11/proxyScan/utils.NewCollector[...] in goroutine 1
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/utils/collect2slice.go:14 +0xd2

goroutine 56 [runnable]:
github.com/google/gopacket.(*eagerPacket).NextDecoder(0xc0008a4160?, {0x10b7d20?, 0xc0002b9268?})
        /root/go/pkg/mod/github.com/google/gopacket@v1.1.19/packet.go:437 +0xe5
github.com/google/gopacket/layers.decodeEthernet({0xc000899080, 0x36, 0x36}, {0x10ccbf0, 0xc0008a4160})
        /root/go/pkg/mod/github.com/google/gopacket@v1.1.19/layers/ethernet.go:122 +0xe6
github.com/google/gopacket.DecodeFunc.Decode(0x4759ba?, {0xc000899080?, 0x40cec5?, 0xc32400?}, {0x10ccbf0?, 0xc0008a4160?})
        /root/go/pkg/mod/github.com/google/gopacket@v1.1.19/decode.go:87 +0x35
github.com/google/gopacket.LayerType.Decode(0xc32400?, {0xc000899080?, 0x41695b?, 0x8fde0?}, {0x10ccbf0?, 0xc0008a4160?})
        /root/go/pkg/mod/github.com/google/gopacket@v1.1.19/layertype.go:95 +0xae
github.com/google/gopacket.(*eagerPacket).initialDecode(0xc0008a4160, {0x10b76e0?, 0x17e20c8?})
        /root/go/pkg/mod/github.com/google/gopacket@v1.1.19/packet.go:453 +0x76
github.com/google/gopacket.NewPacket({0x391a020?, 0x36, 0x18b3620?}, {0x10b76e0, 0x17e20c8}, {0x2?, 0x0?, 0x0?, 0x0?})
        /root/go/pkg/mod/github.com/google/gopacket@v1.1.19/packet.go:682 +0x228
github.com/hdu-dn11/proxyScan/scan/tcpscanner/pcap.(*Scanner).recLoop(0xc000600360, {0x10c3618, 0xc0004dc1e0})
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/scan/tcpscanner/pcap/pcap.go:166 +0x23d
created by github.com/hdu-dn11/proxyScan/scan/tcpscanner/pcap.NewScanner in goroutine 1
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/scan/tcpscanner/pcap/pcap.go:79 +0x497

goroutine 57 [runnable]:
syscall.Syscall(0x1, 0x2, 0xc0004fd380, 0x2b)
        /usr/local/go/src/syscall/syscall_linux.go:69 +0x25
syscall.write(0xc000058180?, {0xc0004fd380?, 0xdac580?, 0xc0001e9c68?})
        /usr/local/go/src/syscall/zsyscall_linux_amd64.go:964 +0x3b
syscall.Write(...)
        /usr/local/go/src/syscall/syscall_unix.go:209
internal/poll.ignoringEINTRIO(...)
        /usr/local/go/src/internal/poll/fd_unix.go:736
internal/poll.(*FD).Write(0xc000058180, {0xc0004fd380, 0x2b, 0x40})
        /usr/local/go/src/internal/poll/fd_unix.go:380 +0x368
os.(*File).write(...)
        /usr/local/go/src/os/file_posix.go:46
os.(*File).Write(0xc00007e030, {0xc0004fd380?, 0x2b, 0x41fa91?})
        /usr/local/go/src/os/file.go:189 +0x51
log.(*Logger).output(0xc0000bc330, 0x0, 0x2, 0xc0001e9f80)
        /usr/local/go/src/log/log.go:245 +0x465
log.Println(...)
        /usr/local/go/src/log/log.go:405
github.com/hdu-dn11/proxyScan/scan.(*Scanner).ScanSocks5.func1()
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/scan/scan.go:75 +0x155
created by github.com/hdu-dn11/proxyScan/scan.(*Scanner).ScanSocks5 in goroutine 1
        /root/go/pkg/mod/github.com/hdu-dn11/proxy!scan@v0.0.0-20240804145125-801fb804c599/scan/scan.go:73 +0x1c6

```

**Comment by BaiMeow** [comment]
少加了个lock，原生map爆了，考虑sync.Map替代

**Comment by BaiMeow** [comment]
solved at 6aa61b11fd09126f3dbcd4bc45769adf4c00175c
